src/gui/sdlmain.cpp

```diff
@@ -4616,6 +4616,31 @@ static void override_wm_class()
 #endif
 }
 
+// Only checks if config file exists and is not empty
+static bool config_file_is_valid(const std_fs::path& path)
+{
+	FILE* file = fopen(path.string().c_str(), "r");
+	if (!file) {
+		return false;
+	}
+
+	constexpr size_t BufferSize = 4096;
+	char buffer[BufferSize];
+	size_t bytes_read;
+	do {
+		bytes_read = fread(buffer, 1, BufferSize, file);
+		for (size_t i = 0; i < bytes_read; ++i) {
+			if (!isspace(buffer[i])) {
+				fclose(file);
+				return true;
+			}
+		}
+	} while (bytes_read == BufferSize);
+
+	fclose(file);
+	return false;
+}
```

Comment:
That's a nice optimal way of checking if the file is empty without rewriting the config parser (which will be refactored anyway) 🎉 

My only suggestion is to move it to `setup.cpp` (and upper camel case the name) as it doesn't really belong in sdlmain.

Reply:
Moved and renamed as you suggested.  I am curious why non-member functions in that file use `MicrosoftCase` as opposed to `snake_case` though.  I thought `snake_case` was our coding style for stand alone functions.

Reply:
To be honest I wasn't thinking about it too deeply, just following the conventions in `setup.cpp` 😄  I guess all functions in `setup.h` would need to be prefixed with `SETUP_`; maybe you do wanna make adjustment quickly too 😄 

```cpp
std_fs::path GetPrimaryConfigPath();
void ParseConfigFiles(const std_fs::path& config_path);

const std::string& GetLanguage();

const char* SetProp(std::vector<std::string>& pvars);
```

Reply:
Actually, I think those should all be members of the `Config` class.  They're all referencing this `control` global (of type `Config`) when they should probably be member functions with the caller referencing the global.

`ConfigFileIsValid` is just a standalone function though.

Reply:
I'm gonna do a small bit of refactoring here.

Reply:
See second commit whenever you get a moment to see if the refactor looks good to you.  I saw several other places we're not following code conventions for stand-alone functions, especially in `cross.cpp` and `support.cpp` but I'm not going to refactor all that in this PR.

Reply:
That's fine man; thanks for doing some additional cleanup!